docs/develop/ethereum-to-ton/tvm-vs-evm.md

```diff
@@ -0,0 +1,86 @@
+
+### TON Virtual Machine (TVM)
+
+- The TON Virtual Machine (TVM) supports a more diverse range of arithmetic operations, including 64-bit, 128-bit, and 256-bit integers, both unsigned and signed, as well as modulo operations. TVM further enhances its arithmetic capabilities with operations like multiply-then-shift and shift-then-divide, which are particularly useful for implementing fixed-point arithmetic. This variety allows developers to select the most efficient arithmetic operations based on the specific requirements of their smart contracts, offering potential optimizations based on data size and type.
+
+## Overflow checks
+
+### Ethereum Virtual Machine (EVM)
+
+- In the EVM, overflow checks are not inherently performed by the virtual machine itself. With the introduction of Solidity 0.8.0, automatic overflow and underflow checks were integrated into the language to enhance security. These checks help prevent common vulnerabilities related to arithmetic operations but require newer versions of Solidity, as earlier versions necessitate manual implementation of these safeguards. 
+
+### TON Virtual Machine (TVM)
+
+- In contrast, TVM automatically performs overflow checks on all arithmetic operations, a feature built directly into the virtual machine. This design choice simplifies the development of smart contracts by inherently reducing the risk of errors and enhancing the overall reliability and security of the code.
+
+## Cryptography and hash functions
+
+### Ethereum Virtual Machine (EVM)
+
+- EVM has support for the Ethereum-specific cryptography scheme, such as the secp256k1 elliptic curve and the keccak256 hash function. Also, EVM does not have built-in support for Merkle proofs, which are cryptographic proofs used to verify the membership of an element in a set.
+
+### TON Virtual Machine (TVM)
+
+- TVM offers support for 256-bit Elliptic Curve Cryptography (ECC) for predefined curves, like Curve25519. It also supports Weil pairings on some elliptic curves, which are useful for fast implementation of zk-SNARKs (zero-knowledge proofs). Popular hash functions like sha256 are also supported, providing more options for cryptographic operations. In addition, TVM can work with Merkle proofs, providing additional cryptographic features that can be beneficial for certain use cases, such as verifying the inclusion of a transaction in a block.
+
+## High-level languages
+
+### Ethereum Virtual Machine (EVM)
+
+- EVM primarily uses Solidity as its high-level language, which is an object-oriented, statically-typed language similar to JavaScript and C++. Also, there are other languages for writing Ethereum smart-contracts such as Vyper, Yul, etc.
+
+### TON Virtual Machine (TVM)
+
+- TVM uses FunC as a high-level language designed for writing TON smart contracts. It is a functional language with static types and support for algebraic data types. FunC compiles to Fift, which in turn compiles to TVM bytecode.
```

Comment:
Tiny nit — FunC is procedural, not functional (IMHO). It doesn't support currying, functions have more than one parameter and functions are not first-class citizens.